daemon/endpoint.go

```diff
@@ -271,7 +271,7 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 				hasSidecarProxy := e.HasSidecarProxy()
 				e.RUnlock()
 
-				if hasSidecarProxy {
+				if hasSidecarProxy && e.HasBPFProgram() {
```

Comment:
Does this need a timeout?

Reply:
Where does it actually block until the BPF program is ready?

Reply:
This is still WIP. I will mark it as ready for review when it is ready.

Reply:
To answer your question - the logic in `endpoint.HasBPFProgram()` returns true if the channel is closed, false otherwise (which is achieved via `select`). It shouldn't be blocking. The line you commented on, 274 is within a for loop. If `e.HasBPFProgram()` returns false, the for loop will iterate again. This blocks until the BPF program is ready.